app/parse.py

```diff
@@ -8,5 +16,35 @@ class Course:
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+def get_course_info_soup(course_soup: BeautifulSoup):
+    course_info_page = course_soup.select_one("a")["href"]
+    course_url = urljoin(BASE_URL, course_info_page)
+
+    course_page = requests.get(course_url).content
```

Comment:
Consider adding error handling for the HTTP request. If the request fails, it could raise an exception, which should be caught and handled appropriately.